tool/jt.rb

```diff
@@ -2094,14 +2109,28 @@ def profile(*args)
     Dir.mkdir(PROFILES_DIR) unless Dir.exist?(PROFILES_DIR)
 
     time = Time.now.strftime('%Y%m%d-%H%M%S')
-    svg_filename = "#{PROFILES_DIR}/flamegraph_#{time}.svg"
+    suffix = ENV['JT_PROFILE_SUFFIX'].yield_self { |s| "_#{s}" if !s.to_s.empty? }
+    svg_filename = "#{PROFILES_DIR}/flamegraph_#{time}#{suffix}.svg"
 
     if stdin
       File.write(svg_filename, STDIN.read)
     else
       require 'benchmark'
       FileUtils.rm_f(svg_filename)
-      run_args = DEFAULT_PROFILE_OPTIONS + ["--cpusampler.OutputFile=#{svg_filename}"] + args
+      run_args = if truffleruby?
+                   DEFAULT_PROFILE_OPTIONS + ["--cpusampler.OutputFile=#{svg_filename}"] + args
+                 else
+                   repo = flamegraph_repo
+                   env = env.merge(
+                     ensure_gem_installed('ruby-prof', '0.18.0'),
+                     ensure_gem_installed('ruby-prof-flamegraph', '0.3.0'),
```

Comment:
this is what i found searching for ruby flamegraphs, so i'm happy to try out another way.
if it didn't need specific gem versions it would make things a lot simpler.
i'll give this a try, thanks!